composer/trainer/trainer_hparams.py

```diff
@@ -107,10 +109,105 @@
+        train_batch_size (int): The batch size to use for training.
+        dataloader (DataloaderHparams): Hparams used for constructing the dataloader which will be used
+            for loading the train dataset and (if provided) the validation dataset.
+        max_duration (str): The maximum duration to train as a str (e.g. ``1ep``, or ``10ba``).
+            Will be converted to a :class:`~composer.core.Time` object.
+
+            .. seealso:: :class:`~composer.core.Time` for more details on time construction.
+        datadir (str, optional): Datadir to apply for both the training and validation datasets. If specified,
+            it will override both ``train_dataset.datadir`` and ``val_dataset.datadir``. (default: ``None``)
+        val_dataset (DatasetHparams, optional): Hparams for constructing the dataset used for evaluation.
+            (default: ``None``)
+
+            .. seealso:: :mod:`composer.datasets` for datasets built into Composer.
+        eval_batch_size (int, optional): The batch size to use for evaluation. Must be provided if one of
+            ``val_dataset`` or ``evaluators`` is set. (default: ``None``)
+        evaluators (List[EvaluatorHparams], optional): Hparams for constructing evaluators to be used during the
+            eval loop. Evaluators should be used when evaluating one or more specific metrics across one
+            or more datasets. (default: ``None``)
+
+            .. seealso:: :class:`~composer.core.evaluator.Evaluator` for more details on evaluators.
+        algorithms (List[AlgorithmHparams], optional): The algorithms to use during training. (default: ``[]``)
+
+            .. seealso:: :mod:`composer.algorithms` for the different algorithms built into Composer.
+        optimizers (OptimizerHparams, optional): The hparams for constructing the optimizer. (default: ``None``)
+
+            .. seealso:: :class:`.Trainer` for the default optimizer behavior when ``None`` is provided.
+
+            .. seealso:: :mod:`composer.optim` for the different optimizers built into Composer.
+        schedulers (List[SchedulerHparams], optional): The learning rate schedulers. (default: ``[]``).
+
+            .. seealso:: :class:`.Trainer` for the default scheduler behavior when ``[]`` is provided.
+
+            .. seealso:: :mod:`composer.optim.scheduler` for the different schedulers built into Composer.
+        device (DeviceHparams): Hparams for constructing the device used for training.
+            (default: ``CPUDeviceHparams``)
+        grad_accum (int, optional): See :class:`.Trainer`.
+        grad_clip_norm (float, optional): See :class:`.Trainer`.
+        validate_every_n_batches (int, optional): See :class:`.Trainer`.
+        validate_every_n_epochs (int, optional): See :class:`.Trainer`.
+        compute_training_metrics (bool, optional): See :class:`.Trainer`.
+        precision (str or Precision, optional): See :class:`.Trainer`.
```

Comment:
yahp doesn't support unions like this...

```suggestion
        precision (Precision, optional): See :class:`.Trainer`.
```